src/components.ts

```diff
@@ -213,7 +215,7 @@ export async function initComponents(): Promise<AppComponents> {
   const communityEvents = await createCommunityEventsComponent({ config, logs, fetcher, redis })
 
   // AI Compliance components
-  const aiCompliance = await createAIComplianceComponent({ config, logs, featureFlags, metrics })
+  const aiCompliance = await createAIComplianceComponent({ config, logs, featureFlags, metrics, memoryCache })
```

Comment:
Shouldn't we use Redis instead, so the cache doesn't get flushed when the service restarts? Also, a good refactor would be:
```typescript
const memoryStorage = redisHost ? createRedis(...) : createInMemoryCache(...)
```